lighthouse-core/test/config/budget-test.js

```diff
@@ -59,70 +61,153 @@ describe('Budget', () => {
+      budgetJson[0].timings[0].metric = 'lastMeaningfulPaint';
+      assert.throws(_ => Budget.initializeBudget(budgetJson), /Invalid timing metric/);
     });
 
     it('throws when an invalid budget is supplied', () => {
-      budget[0].timings[0].budget = '100KB';
-      assert.throws(_ => Budget.initializeBudget(budget), /Invalid budget/);
+      budgetJson[0].timings[0].budget = '100KB';
+      assert.throws(_ => Budget.initializeBudget(budgetJson), /Invalid budget/);
     });
 
     it('throws when an invalid tolerance is supplied', () => {
-      budget[0].timings[0].tolerance = '100ms';
-      assert.throws(_ => Budget.initializeBudget(budget), /Invalid tolerance/);
+      budgetJson[0].timings[0].tolerance = '100ms';
+      assert.throws(_ => Budget.initializeBudget(budgetJson), /Invalid tolerance/);
     });
 
     it('throws when an invalid property is supplied', () => {
-      budget[0].timings[0].device = 'Phone';
-      assert.throws(_ => Budget.initializeBudget(budget), /[device]/);
+      budgetJson[0].timings[0].device = 'Phone';
+      assert.throws(_ => Budget.initializeBudget(budgetJson), /[device]/);
+    });
+
+    it('throws when "time-to-interactive is supplied', () => {
+      budgetJson[0].timings[0].metric = 'time-to-interactive';
+      assert.throws(_ => Budget.initializeBudget(budgetJson), /Invalid timing metric/);
+    });
+  });
+
+  describe('path validation', () => {
+    it('initializes correctly', () => {
+      const budgetArr = [{}];
+
+      budgetArr[0].path = '/';
+      assert.equal(Budget.initializeBudget(budgetArr)[0].path, '/');
+
+      budgetArr[0].path = '/*';
+      assert.equal(Budget.initializeBudget(budgetArr)[0].path, '/*');
+
+      budgetArr[0].path = '/fish*.php';
+      assert.equal(Budget.initializeBudget(budgetArr)[0].path, '/fish*.php');
+
+      budgetArr[0].path = '/*.php$';
+      assert.equal(Budget.initializeBudget(budgetArr)[0].path, '/*.php$');
+    });
+  });
+
+  describe('path validation', () => {
+    it('throws error if an invalid path is used', () => {
+      const budget = {};
+
+      budget.path = '';
+      assert.throws(_ => Budget.initializeBudget(budget), /[A valid path]/);
+
+      budget.path = 'cat';
+      assert.throws(_ => Budget.initializeBudget(budget), /[Invalid path]/);
+
+      budget.path = '/cat*cat*cat';
+      assert.throws(_ => Budget.initializeBudget(budget), /[Invalid path]/);
+
+      budget.path = '/cat$html';
+      assert.throws(_ => Budget.initializeBudget(budget), /[Invalid path]/);
+    });
+
+    it('matches root', () => {
+      assert.ok(Budget.urlMatchesPattern('https://google.com', '/'));
+      assert.ok(Budget.urlMatchesPattern('https://google.com', '*'));
+    });
+
+    it('ignores origin', () => {
+      assert.equal(Budget.urlMatchesPattern('https://yt.com/videos?id=', '/videos'), true);
+      assert.equal(Budget.urlMatchesPattern('https://go.com/dogs', '/go'), false);
+    });
+
+    it('is correct', () => {
```

Comment:
would you mind splitting out the 4 cases of combined `*`/`$` usage so it's a little clearer we've got coverage for all of them?